ui/src/app/i18n-strings.ts

```diff
@@ -0,0 +1,31 @@
+import {InjectionToken} from '@angular/core';
+
+import {LanguageCode, StringAssets} from './i18n-helper';
+
+// This type define makes sure the language code used in ALL_LANGUAGE_ASSETS is
+// expected and no missing field in any translation.
+type LanguageAssets = Partial<Record<LanguageCode, StringAssets>>;
+
+/**
+ * Strings used in UI in all supported languages.
+ * We only provide English string. To add a new lanuage, do the following:
+ * 1. copy and paste everything in 'en'.
+ * 2. change all engilish strings to the correct translation.
+ * 3. change the key to the correponding language code string. The string must
+ * be one of the defined LanguageCode in i18-helper.ts.
+ */
+export const ALL_LANGUAGE_ASSETS: LanguageAssets = {
+  // English
+  'en': {
```

Comment:
Should this use LanguageCode['English(US)']?

Reply:
Yeah. I would have to do something like this:
{
  [LanguageCode['English(US)']]: {},
}
which is not very readable compare to 'en'. The compiler also recognize 'en' as LanguageCode['English(US)'] in this case. So I used the simpler version.